### PR TITLE
Use Text component in mobile DMs

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Figured I'd go ahead and change all the `Text` usages in mobile chats to our `Text` component, after all the incorrect fonts we caught.

Also caught an erroneous `marginTop` that was causing incorrect layout in `ChatUserListItem`.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Local ios stage
Simulator Screenshot - iPhone 14 Pro - 2023-07-14 at 16 27 51

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/chat-screen/ChatMessageSeparator.tsx [hunk 1]
```diff
@@ -34,7 +32,14 @@ export const ChatMessageSeparator = ({ content }: { content: string }) => {
   return (
     <View style={styles.unreadTagContainer}>
       <View style={styles.unreadSeparator} />
-      <Text style={styles.unreadTag}>{content}</Text>
+      <Text
```
Comment (schottra) [inline packages/mobile/src/screens/chat-screen/ChatMessageSeparator.tsx:35]: Is there a default color on Text if you don't specify one? I saw below that other usage included a `color` prop.

File: packages/mobile/src/screens/chat-screen/ChatUserListItem.tsx [hunk 2]
```diff
@@ -258,7 +250,15 @@ export const ChatUserListItem = ({ userId }: ChatUserListItemProps) => {
                 )}
               </View>
               {user.does_follow_current_user && canCreateChat ? (
-                <Text style={styles.followsYouTag}>{messages.followsYou}</Text>
+                <Text
+                  fontSize='xxs'
+                  weight='heavy'
+                  color='neutralLight4'
```
Comment (schottra) [inline packages/mobile/src/screens/chat-screen/ChatUserListItem.tsx:256]: Worth thinking about going forward if we should add the semantic variants of text color and use those instead (i.e. `textSubdued`)

**Review by schottra — APPROVED** [review]
Good cleanup!

File: packages/mobile/src/screens/chat-screen/ChatMessageSeparator.tsx (hunk 1 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ChatMessageSeparator.tsx:35]: yep it defaults to `neutral`.

**Pull request merged**